pkg/apis/serving/k8s_validation.go

```diff
@@ -505,15 +504,46 @@ func validateContainersPorts(containers []corev1.Container) (corev1.ContainerPor
 }
 
 // validateSidecarContainer validate fields for non serving containers
-func validateSidecarContainer(ctx context.Context, container corev1.Container, volumes map[string]corev1.Volume) (errs *apis.FieldError) {
+func validateSidecarContainer(ctx context.Context, container corev1.Container, volumes map[string]corev1.Volume, mainContainerPort corev1.ContainerPort) (errs *apis.FieldError) {
+	// Checks if liveness/readiness probes are the same as the main container port
```

Comment:
I think either is fine having this validation or not, because main container's probe port can map to the sidecar container port, also there are some other ports the sidecar probe cannot use such as `QUEUE_SERVING_PORT(8012)`.

But there is no harm to have this validation so let's keep it as it is.

Reply:
Thank you for reviewing the PR and for your insights on the port mapping between the main container and the sidecar container, as well as limitations like the QUEUE_SERVING_PORT.

I'll keep it as it is for now based on your recommendation.